specs/protocol/revenue-sharing.md

```diff
@@ -0,0 +1,48 @@
+# Revenue Sharing
+
+Revenue sharing is the process by which chains in the superchain contribute a portion of their revenue or profit to the Optimism collective. They do this in return for support with the OP Stack and other benefits. 
+
+
+## Definitions
+| Term   | Name         | Definition  |
+| -------|--------------| ----------- |
+| $d$ | L1 Data Fee Revenue    | ETH transferred to the sequencer with L2 transactions to cover estimated L1 Data Fees(see below). Accumulates to `L1FeeVault`.
+| $e$ | L1 Data Fee Expenditure| ETH spent by the batcher on L1 to make transaction data available
+| $b$ | L2 Base Gas Revenue         | Base fee portion of L2 Gas Fee (ETH transferred to the sequencer to cover execution of L2 transactions). Accumulates to `BaseFeeVault`.
+| $p$ | L2 Priority Gas Revenue         | Priority fee portion of L2 Gas Fee (ETh transferred to the sequencer to cover execution of L2 transactions). Accumulates to `SequencerFeeVault`.
+| $g$ | L2 Gas Revenue         | $b+p$
+| $r$ | Sequencer Revenue      | $e + g$
+| $p$ | Sequencer Profit       | $e + g - d$
+| $s$ | Revenue share due to Optimism Collective | $\max(0.15r,0.025p)$
+
+## `RevenueSharer` predeploy
+Revenue sharing is achieved through an L2 contract `RevenueSharer`. This is the implementation of a proxy contract `RevenueSharerProxy` with address
+
+```
+0x4200000000000000000000000000000000000022
+```
+
+### Deploying `RevenueSharer`
```

Comment:
Thoughts on `RevenueShare` vs `RevenueSharer`? Seems a bit easier to say and gets the same point across

Reply:
I don't have super strong opinion. I think having the `-er` follows existing predeploys such as `L2ToL1MessagePasser `, `L2CrossDomainMessenger` and so on. And we do already have some hard-to-say names such as `OptimismMintableERC721Factory`. 

Other options are `RevShare` and `RevSharer`. Base are using something they call a `FeeDisburser`.

Reply:
I have a preference to just call it `RevenueShare` to make it easier to say. the "revenue share predeploy" vs the "revenue sharer predeploy"

Reply:
bump

Reply:
Slack poll so far indicating a preference for `RevenueSharer`. I'm going to to continue using this name for now, it should be easy to replace down the line if necessary.